proxy/src/main/java/com/wavefront/agent/listeners/tracing/CustomTracingPortUnificationHandler.java

```diff
@@ -0,0 +1,173 @@
+   * @param healthCheckManager        shared health check endpoint handler.
+   * @param traceDecoder              trace decoders.
+   * @param spanLogsDecoder           span logs decoders.
+   * @param preprocessor              preprocessor.
+   * @param handlerFactory            factory for ReportableEntityHandler objects.
+   * @param sampler                   sampler.
+   * @param alwaysSampleErrors        always sample spans with error tag.
+   * @param traceDisabled             supplier for backend-controlled feature flag for spans.
+   * @param spanLogsDisabled          supplier for backend-controlled feature flag for span logs.
+   * @param wfSender                  sender to send trace to Wavefront.
+   * @param traceDerivedCustomTagKeys custom tags added to derived RED metrics.
+   */
+  public CustomTracingPortUnificationHandler(
+      String handle, TokenAuthenticator tokenAuthenticator, HealthCheckManager healthCheckManager,
+      ReportableEntityDecoder<String, Span> traceDecoder,
+      ReportableEntityDecoder<JsonNode, SpanLogs> spanLogsDecoder,
+      @Nullable Supplier<ReportableEntityPreprocessor> preprocessor,
+      ReportableEntityHandlerFactory handlerFactory, Sampler sampler, boolean alwaysSampleErrors,
+      Supplier<Boolean> traceDisabled, Supplier<Boolean> spanLogsDisabled,
+      @Nullable WavefrontSender wfSender, Set<String> traceDerivedCustomTagKeys) {
+    this(handle, tokenAuthenticator, healthCheckManager, traceDecoder, spanLogsDecoder,
+        preprocessor, handlerFactory.getHandler(HandlerKey.of(ReportableEntityType.TRACE, handle)),
+        handlerFactory.getHandler(HandlerKey.of(ReportableEntityType.TRACE_SPAN_LOGS, handle)),
+        sampler, alwaysSampleErrors, traceDisabled, spanLogsDisabled, wfSender,
+        traceDerivedCustomTagKeys);
+  }
+
+  @VisibleForTesting
+  public CustomTracingPortUnificationHandler(
+      String handle, TokenAuthenticator tokenAuthenticator, HealthCheckManager healthCheckManager,
+      ReportableEntityDecoder<String, Span> traceDecoder,
+      ReportableEntityDecoder<JsonNode, SpanLogs> spanLogsDecoder,
+      @Nullable Supplier<ReportableEntityPreprocessor> preprocessor,
+      final ReportableEntityHandler<Span, String> handler,
+      final ReportableEntityHandler<SpanLogs, String> spanLogsHandler, Sampler sampler,
+      boolean alwaysSampleErrors, Supplier<Boolean> traceDisabled,
+      Supplier<Boolean> spanLogsDisabled, @Nullable WavefrontSender wfSender,
+      Set<String> traceDerivedCustomTagKeys) {
+    super(handle, tokenAuthenticator, healthCheckManager, traceDecoder, spanLogsDecoder,
+        preprocessor, handler, spanLogsHandler, sampler, alwaysSampleErrors, traceDisabled, spanLogsDisabled);
+    this.wfSender = wfSender;
+    this.discoveredHeartbeatMetrics = new ConcurrentHashMap<>();
+    this.traceDerivedCustomTagKeys = traceDerivedCustomTagKeys;
+
+    if (wfSender != null) {
+      wfInternalReporter = new WavefrontInternalReporter.Builder().
```

Comment:
Might be a good idea to actually pass a WavefrontInternalReporter as a constructor argument (as well as starting it outside of the constructor). The way it's implemented right now it's quite difficult to create a mock for the reporter for testing purposes.